packages/components/src/components/Select/Select.tsx

```diff
@@ -110,6 +114,29 @@ export const SelectInput = (
             itemToString: (item) => (item ? item.label : ''),
         });
 
+    const { id: toggleButtonId, ...toggleButtonProps } = useMemo(
+        () =>
+            getToggleButtonProps({
+                'aria-label': ariaLabel,
+                ...(forwardedRef ? { ref: forwardedRef } : {}),
+            }),
+        [ariaLabel, forwardedRef, getToggleButtonProps],
+    );
+
+    useLayoutEffect(() => {
+        const labelClickHandler = () => document.getElementById(toggleButtonId)?.click();
+
+        if (!labelId) {
+            return;
+        }
+
+        document.getElementById(labelId)?.addEventListener('click', labelClickHandler);
+
+        return () => {
+            document.getElementById(labelId)?.removeEventListener('click', labelClickHandler);
+        };
+    }, [labelId, toggleButtonId]);
```

Comment:
Pls add it in the label component
```tsx
data-test-id={dataTestId}
onClick={(event) => {
    // Add support of Select component, Radix only allows native `select`
    // but we use `div[role=combobox]` because of downshift
    // https://github.com/radix-ui/primitives/blob/6e75e117977c9e6ffa939e6951a707f16ba0f95e/packages/react/label/src/label.tsx#L22
    const targetId = (event.target as HTMLElement).getAttribute('for');
    const target = targetId ? document.getElementById(targetId) : null;

    if (target && target.getAttribute('role') === 'combobox') {
        event.preventDefault();
        event.stopPropagation();
        target.click();
    }
}}
{...props}
```

Reply:
Not really a fan of either solution. But okay, this makes it closest to the native behaviour for the developers.